phf_shared/src/foldhash.rs

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) 2024 Orson Peters
+// Distributed with permission under the zlib license.
+// Minor modifications made to remove any platform dependant code.
```

Comment:
I'm not sure but does this affect our current license, MIT? I'd like to remain MIT as changing license is a hassle in general.

Reply:
Why does PR inline the foldhash implementation rather than depending on the `foldhash` crate? That would solve the licensing issue...